src/redux/reducers/authReducer.ts

```diff
@@ -0,0 +1,35 @@
+import { AuthActionsTypes } from '../actions/authActions';
+import { SignInPayload } from '../actions/authActions/successSignInAction';
+
+export type AuthReducerState = {
+  userToken: SignInPayload['userToken'] | null;
+  user: SignInPayload['user'];
+};
+
+const initialState: AuthReducerState = {
+  userToken: null,
+  user: undefined,
+};
+
+export const authReducer = (
+  state = initialState,
+  action: any,
+): AuthReducerState => {
+  switch (action.type) {
+    case AuthActionsTypes.SIGN_IN: {
+      const { user, userToken }: SignInPayload = action.payload;
+      return {
+        ...state,
+        userToken,
+        user,
+      };
+    }
+    case AuthActionsTypes.SIGN_OUT: {
+      return {
+        ...state,
```

Comment:
are you sure that need to save user after signout? 

usually on SIGN_OUT action set initialState to reducer

Reply:
i fix it

Reply:
i think @SHEMPEL133 means that we have to return initial state, not just use {...state, blablabla}